active/XamlServiceProvider/XamlServiceProviderApi.md

```diff
@@ -0,0 +1,272 @@
+{
+    public string Format { get; set; } = "";
+    protected override object ProvideValue()
+    {
+        return DateTime.Now.ToString(Format);
+    }
+}
+```
+
+you can do this in Xaml markup:
+
+```xml
+<TextBlock>
+    Today is:  
+    <Run Text="{local:CurrentDate Format=d}"/>
+</TextBlock>
+```
+
+On an en-US machine, this will produce something like:
+ 
+![MarkupExtension sample](MESample.png)
+
+# Examples
+
+## IProvideValueTarget
+
+The following example shows a custom markup extension whose provided value changes based on the specific property being targeted by the custom markup extension. In this case, the developer wants something that will automatically use an AcrylicBrush for the [Control.Background](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Control.Background) or [Border.Background](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Border.Background) properties, but uses a [SolidColorBrush](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) for all other properties. The ProvideValue implementation deterimines what value to return by using the IProvideValueTarget it gets from the IXamlServiceProvider parameter passed to ProvideValue.
+
+```cs
+public class BrushSelectorExtension : MarkupExtension
+{
+    public Color Color { get; set; }
+
+    protected override object ProvideValue(IXamlServiceProvider serviceProvider)
+    {
+        Brush brushToReturn = new SolidColorBrush() { Color = Color }; ;
+
+        var provideValueTarget = serviceProvider.GetService(typeof(IProvideValueTarget))
+            as IProvideValueTarget;
+        if (provideValueTarget.TargetProperty is ProvideValueTargetProperty targetProperty)
+        {
+            if (targetProperty.Name == "Background"
+                && (targetProperty.DeclaringType == typeof(Control) 
+                    || targetProperty.DeclaringType == typeof(Border)))
+            {
+                brushToReturn = new AcrylicBrush()
+                    { TintColor = Color, TintOpacity = 0.75 };
+            }
+        }
+
+        return brushToReturn;
+    }
+}
+```
+
+```xml
+<StackPanel>
+    <Button Foreground="{local:BrushSelector Color=Blue}" 
+            Background="{local:BrushSelector Color=Gold}">
+        Go bears!
+    </Button>
+    <Rectangle x:Name="SolidColor" Fill="{local:BrushSelector Color=Green}" />
+</StackPanel>
+```
+
+## IRootObjectProvider
+
+> Some more realistic example?
+
+The following example shows a custom markup extension that returns a string describing the object at the root of the markup. It does this by getting the IRootObjectProvider from the IXamlServiceProvider parameter passed to ProvideValue.
+
+With this markup extension:
+
+```cs
+public class TestMarkupExtension : MarkupExtension
+{
+    public override object ProvideValue(IXamlServiceProvider serviceProvider)
+    {
+        var target = serviceProvider.GetService(typeof(IRootObjectProvider)) as IRootObjectProvider;

```

Comment:
`target` is a bad name for this variable because it's not the target of the markup extension.